exporter/clickhouseexporter/config.go

```diff
@@ -145,6 +172,48 @@ func (cfg *Config) shouldCreateSchema() bool {
 	return cfg.CreateSchema
 }
 
+func (cfg *Config) ValidateMetricTableNames() error {
+	if len(cfg.MetricsTableName) == 0 {
+		return nil
+	}
+	if cfg.areTableNamesSet() {
+		return fmt.Errorf("Warning: 'metrics_table_name' is deprecated, use 'metrics_tables' instead")
+	}
+	return nil
+}
+
+func (cfg *Config) buildTableNames() {
+	tableName := defaultMetricTableName
+
+	if len(cfg.MetricsTableName) != 0 && !cfg.areTableNamesSet() {
+		tableName = cfg.MetricsTableName
+	}
+
+	if len(cfg.MetricsTables.Gauge) == 0 {
+		cfg.MetricsTables.Gauge = tableName + defaultGaugeSuffix
+	}
+	if len(cfg.MetricsTables.Sum) == 0 {
+		cfg.MetricsTables.Sum = tableName + defaultSumSuffix
+	}
+	if len(cfg.MetricsTables.Summary) == 0 {
+		cfg.MetricsTables.Summary = tableName + defaultSummarySuffix
+	}
+	if len(cfg.MetricsTables.Histogram) == 0 {
+		cfg.MetricsTables.Histogram = tableName + defaultHistogramSuffix
+	}
+	if len(cfg.MetricsTables.ExponentialHistogram) == 0 {
+		cfg.MetricsTables.ExponentialHistogram = tableName + defaultExpHistogramSuffix
+	}
+}
+
+func (cfg *Config) areTableNamesSet() bool {
```

Comment:
This function along with `ValidateMetricTableNames` seems excessively complicated. I believe the other function (`buildTableNames`) can be reworked to solve everything. I don't think we need to emit a warning that the field is deprecated, a note in the README will work fine.

Let's try to simplify all of these functions and checks into one if possible. We simply need to populate table names for the `metricsExporter`. One function should be enough to encapsulate the logic for the new struct + falling back to the deprecated field. Then we can call this one function in the config parser, same as you have now.